doc/api/metrics.md

```diff
@@ -0,0 +1,1040 @@
+
+```mjs
+import { counter, timer, statsdStream } from 'node:metrics';
+import { createWriteStream } from 'node:fs';
+
+// Create a counter metric
+const apiCalls = counter('api.calls', { service: 'web' });
+
+// Create a timer factory
+const requestTimer = timer('api.request.duration', { service: 'web' });
+
+// Export metrics to StatsD format
+const statsd = statsdStream();
+statsd.pipe(createWriteStream('metrics.log'));
+
+// Use metrics in your application
+function handleRequest(req, res) {
+  const timer = requestTimer.create({ endpoint: req.url });
+
+  apiCalls.increment();
+
+  // Process request...
+
+  timer.stop();
+}
+```
+
+```cjs
+const { counter, timer, statsdStream } = require('node:metrics');
+const { createWriteStream } = require('node:fs');
+
+// Create a counter metric
+const apiCalls = counter('api.calls', { service: 'web' });
+
+// Create a timer factory
+const requestTimer = timer('api.request.duration', { service: 'web' });
+
+// Export metrics to StatsD format
+const statsd = statsdStream();
+statsd.pipe(createWriteStream('metrics.log'));
+
+// Use metrics in your application
+function handleRequest(req, res) {
+  const timer = requestTimer.create({ endpoint: req.url });
+
+  apiCalls.increment();
+
+  // Process request...
+
+  timer.stop();
+}
+```
+
+## Metric Types
+
+### `metrics.counter(name[, meta])`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* `name` {string} The name of the counter metric.
+* `meta` {Object} Optional metadata to attach to all reports.
+* Returns: {metrics.Counter}
+
+Creates a counter metric that tracks cumulative values.
+
+```mjs
+import { counter } from 'node:metrics';
+
+const errorCount = counter('errors.total', { component: 'database' });
+
+errorCount.increment();     // Increment by 1
+errorCount.increment(5);    // Increment by 5
+errorCount.decrement(2);    // Decrement by 2
+```
+
+
+### `metrics.gauge(name[, meta])`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* `name` {string} The name of the gauge metric.
+* `meta` {Object} Optional metadata to attach to all reports.
+* Returns: {metrics.Gauge}
+
+Creates a gauge metric that represents a single value at a point in time.
+
+```mjs
+import { gauge } from 'node:metrics';
+import { memoryUsage } from 'node:process';
+
+const memory = gauge('memory.usage.bytes');
+
+memory.reset(memoryUsage().heapUsed);
+memory.applyDelta(1024);  // Add 1024 to current value
+```
+
+### `metrics.meter(name, interval[, meta])`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* `name` {string} The name of the meter metric.
+* `interval` {number} The time window in milliseconds for rate calculation.
+* `meta` {Object} Optional metadata to attach to all reports.
+* Returns: {metrics.Meter}
+
+Creates a meter metric that measures the rate of events over time.
+
+```mjs
+import { meter } from 'node:metrics';
+
+const requestRate = meter('requests.rate', 60000); // 1 minute window
+
+requestRate.mark();     // Mark one event
+requestRate.mark(10);   // Mark 10 events
+```
```

Comment:
It's unclear from the documentation here how a `meter` differs from a `counter` and what the semantics of calling `mark(...)` multiple times would be. That is, what is the value of the meter after this example.